src/common/hist_util.cu

```diff
@@ -45,7 +45,7 @@ __global__ void find_cuts_k
     isample = max(0, min(isample, nsamples - 1));
   }
   // repeated values will be filtered out on the CPU
-  bst_float rmin = isample > 0 ? cum_weights[isample - 1] : 0;
+  bst_float rmin = isample > 0 ? cum_weights[isample - 1] : 0.0f;
```

Comment:
I'm confused. What's the difference between `0` and `0.0f`?

Reply:
I’ve looked into it in more detail. It appears to be an issue of thrust. I am following up with the thrust team.